goat.test.ts

```diff
@@ -247,6 +247,23 @@ describe("unit: streaming RPCs", () => {
 
     beforeEach(() => {
         vi.useFakeTimers();
+
+        // Listen to unhandled rejections during tests (e.g. uncaught Promise exceptions).
+        // Note, to fail the test, we assert a mock function not to be called,
+        // instead of asserting directly in the event listener, to be in the test context.
+        const unhandledRejectionFn = vi.fn();
+        const p = process.addListener("unhandledRejection", unhandledRejectionFn);
+
+        // Remove event listener and expect no unhandled rejections
+        return () => {
+            p.removeListener("unhandledRejection", unhandledRejectionFn);
+
+            // Expect any unhandled rejections do not have a defined reason
+            const mockCallArgs = unhandledRejectionFn.mock.lastCall;
+            if (mockCallArgs) {
```

Comment:
I added the `if` stage to get the error message in the `AssertionError` - we get `AssertionError: expected Error: cannot write, WritableIterable alr… { …(4) } to be undefined` instead of `AssertionError: expected [ …(2) ] to be undefined`. I've added a comment and made this a bit more explicit